src/profile-handler.cc

```diff
@@ -382,7 +388,11 @@ ProfileHandler::ProfileHandler()
   // Install the signal handler.
   struct sigaction sa;
   sa.sa_sigaction = SignalHandler;
+#ifdef SA_RESTART
```

Comment:
perhaps easier and cleaner is something like that instead (outside of function body)

```
#ifndef SA_RESTART
#define SA_RESTART 0
#endif
```